2025 review blog (#212)
cucumber/website: blog/2026-02-24-cucumber-in-2025-year-in-review.md

### Cucumber JS

In 2025, we've made 8 releases for Cucumber-JS. These were mostly bug fixes,
and small features, but we did make some architectural moves to enable bigger
things. Some notable changes:

* Alexandru Gologan contributed [support for execution sharding](https://github.com/cucumber/cucumber-js/pull/2600).
* Configuration files [can be written in TypeScript](https://github.com/cucumber/cucumber-js/pull/2709).
* You can now write [plugins to augment built-in behaviour](https://github.com/cucumber/cucumber-js/blob/main/docs/plugins.md)

We're also working on [`@cucumber/node`](https://github.com/cucumber/cucumber-node)
built around the [Node.js test runner](https://nodejs.org/api/test.html). It's
still in the pre-1.0.0 phase, so APIs and behaviour might change. The stable
canonical implementation of Cucumber for JavaScript continues to be
[@cucumber/cucumber](https://github.com/cucumber/cucumber-js) for now.

### Cucumber Ruby

In 2025 Cucumber-Ruby turned Seventeen! Which means it is now
[old enough to drive](https://github.com/cucumber/cucumber-ruby/commits/f3292f4023a707099d02602b2bd6c4ca3cec6820)

> in some parts of the world. It also saw 5 releases including v10 which adds

Reviewer: ```suggestion
in some parts of the world. It also saw 5 releases including the v10 version which began to lay the groundwork for a lot of architectural changes
```

* We had [a contribution](https://github.com/cucumber/cucumber-jvm/pull/3124)
  from Stefan Gasterstädt to make it possible to create locale sensitive
  parameter transformers. This looked quite complicated at the start, but the
  actual implementation turned out to be incredibly elegant.

* Julien Kronegg contributed several<sup>
  [1](https://github.com/cucumber/gherkin/pull/372),
  [2](https://github.com/cucumber/cucumber-jvm/pull/2971),
  [3](https://github.com/cucumber/cucumber-jvm/pull/2703),
  [4](https://github.com/cucumber/gherkin/pull/445)</sup> performance
  improvements.

* The Cucumber JUnit Platform Engine now [supports rerun files](https://github.com/cucumber/cucumber-jvm/pull/3057).
  While the solution comes with a bunch of jankiness, it should smooth out the
  upgrade path from JUnit 4 to Junit 5+.

* With JUnit 4 entering maintenance mode [cucumber-junit](https://github.com/cucumber/cucumber-jvm/tree/main/cucumber-junit)
  has been deprecated in favor of [cucumber-junit-platform-engine](https://github.com/cucumber/cucumber-jvm/tree/main/cucumber-junit-platform-engine).

### Cucumber JS

In 2025, we've made 8 releases for Cucumber-JS. These were mostly bug fixes,

> and small features. Other than that, lots of architectural Tetris to unblock

Reviewer 2: ```suggestion
and small features, but we did make some architectural moves to enable bigger
```

things. Some notable changes:

* Alexandru Gologan contributed [support for execution sharding](https://github.com/cucumber/cucumber-js/pull/2600).
* Configuration files [can be written in TypeScript](https://github.com/cucumber/cucumber-js/pull/2709).

> We're also working on a [`cucumber-node`](https://github.com/cucumber/cucumber-node)

Reviewer 2: ```suggestion
We're also working on [`@cucumber/node`](https://github.com/cucumber/cucumber-node)
```

built around the [Node.js test runner](https://nodejs.org/api/test.html). It's
still in the pre-1.0.0 phase, so APIs and behaviour might change. The stable
canonical implementation of Cucumber for JavaScript continues to be
[@cucumber/cucumber](https://github.com/cucumber/cucumber-js) for now.

> <!-- TODO: What about user author plugins -->

Reviewer 2: ```suggestion
```

Yep, I'll add this in the list above.

### Cucumber Ruby

In 2025 Cucumber-Ruby turned Seventeen! Which means it is now
[old enough to drive](https://github.com/cucumber/cucumber-ruby/commits/f3292f4023a707099d02602b2bd6c4ca3cec6820)
in some parts of the world. It also saw 5 releases including v10 which adds
support for Ruby 3.5+.

### Cucumber Messages

[Cucumber Messages](https://github.com/cucumber/messages) is a schema to
describe test results and other information from Cucumber in a new-line
delimited JSON format. The first commit was done in 2018, and we are now at the
point that we can dogfood our own implementation and build up the documentation
and tooling through that.

For users the first results can be seen in the
[pretty formatter](https://github.com/cucumber/pretty-formatter)
and [HTML formatter](https://github.com/cucumber/html-formatter) which now
create reports with more fidelity.

> <!-- TODO: Do we want to include this? -->

Reviewer 2: We could omit until we have more to say, yeah.

In related news, the [reports.cucumber.io](https://reports.cucumber.io) service
is back up and running after the transition from SmartBear. For now without

> permanent storage. We're still figuring out where to take this next.

Author: ```suggestion
```

### Cucumber JSON Report in maintenance-mode

With the internal adoption of messages now in progress we have now also
officially put the Cucumber JSON report in maintenance mode. The format is still

> supported but will not receive any changes. We don't expect this will any

Reviewer 2: ```suggestion
supported but will not receive any changes. We don't expect this will have any
```

significant impact, in practice the format had already been ossified.

Putting the format in maintenance mode meant that we had to pin down the
exact format of the report in the form of the [cucumber-json-schema](https://github.com/cucumber/cucumber-json-schema).
Unfortunately there been a significant drift between implementations and each
flavor will need its own schema. So this is very much a work in progress, but
for Cucumber-JVM at least that work is now completed. 

### Reqnroll

While [Reqnroll](https://reqnroll.net/) isn't under Cucumber umbrella, I would

> be amiss if I didn't find a way to highlight Chris Rudolphi contributions.

Reviewer 2: ```suggestion
be amiss if I didn't find a way to highlight Chris Rudolphi's contributions.
```

Chris has been driving the adoption of Cucumber messages in Reqnroll and as part

> of that made numerous contributions to Cucumber. Including .Net implementations

Reviewer 2: ```suggestion
of that made numerous contributions to Cucumber. Including .NET implementations
```

of tag-expressions, html-formatter and compatibility-kit.

## Organizational stuff

We created a [governance repository](https://github.com/cucumber/governance/) to
describe who we are and how we are organized and operate. This was rather boring
to do but also essential to acquire funding and sponsorships.

Additionally, we have set up a [security policy](https://github.com/cucumber/.github/blob/main/SECURITY.md)
to let people know how to contact us in case of security issues.

## Newsletter

We're finally reviving our mailing list. If you're reading this in your email
client, that means we've got it all working again.

If not, you can subscribe [here](https://cucumber.io/community).

## Finances

Cucumber is currently funded through donations from Open Collective, GitHub
Sponsors, and Thanks.dev. We have also partnered with Tidelift and collect some

> add revenue through Ethical Ads. In total, we received $65,545.63 with the

Reviewer 2: ```suggestion
ad revenue through Ethical Ads. In total, we received $65,545.63 with the
```

majority coming from Tidelift.

| Source          | 2025 Income |
|-----------------|------------:|
| Tidelift        |  $52,900.00 |
| Open Collective |   $7,784.50 |

> | GitHub Sponors  |   $4,047.96 |

Reviewer 2: ```suggestion
| GitHub Sponsors |   $4,047.96 |
```

| Ethical Ads     |     $768.50 |
| Thanks.dev      |      $44.67 |
|                 | --------- + |
| Total           |  $65,545.63 |

> On the expense side we spend $69.960,28 resulting in a small loss of $4,414.65

Reviewer 2: ```suggestion
On the expense side we spent $69.960,28 resulting in a small loss of $4,414.65
```

> Fortunately we already build up some buffer in 2024 to absorb the loss. 

Reviewer 2: ```suggestion
Fortunately we already built up some buffer in 2024 to absorb the loss. 
```
